kubernetes/basic.go

```diff
@@ -0,0 +1,34 @@
+package kubernetes
+
+import (
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// RestConfig from the kubeconfig
+func RestConfig() (restConfig *rest.Config) {
+	// Get a rest.Config from the kubeconfig file.  This will be passed into all
+	// the client objects we create.
+	restConfig, err := kubeconfig().ClientConfig()
+	if err != nil {
+		panic(err)
+	}
+
+	return
+}
+
+// Namespace from the kubeconfig
+func Namespace() (namespace string) {
+	namespace, _, err := kubeconfig().Namespace()
+	if err != nil {
+		panic(err)
```

Comment:
Please avoid using panic(). We have a logging framework that can do the same (use the Fatal error level) and provide more meaningful errors for the user. You can also add fields to the logging message, similar to metadata in an ELK stack (e.g. what was the value of the namespace parameter)

Reply:
e.g. `log.WithField("namespace", namespace).WithError(err).Fatal("Could not get namespace from Kubeconfig")` or whatever the reason for the fail might be.

Reply:
You are totally right, improved that on the whole codebase.